internal/util/importutilv2/parquet/util.go

```diff
@@ -105,77 +93,164 @@ func CreateFieldReaders(ctx context.Context, fileReader *pqarrow.FileReader, sch
-		return typeutil.IsFloatingType(dst) || dst == schemapb.DataType_Int64
+		return &arrow.Int64Type{}, nil
 	case schemapb.DataType_Float:
-		if isList && dst == schemapb.DataType_FloatVector {
-			return true
-		}
-		return typeutil.IsFloatingType(dst)
+		return &arrow.Float32Type{}, nil
 	case schemapb.DataType_Double:
-		if isList && dst == schemapb.DataType_FloatVector {
-			return true
-		}
-		return dst == schemapb.DataType_Double
-	case schemapb.DataType_String, schemapb.DataType_VarChar:
-		return typeutil.IsStringType(dst) || typeutil.IsJSONType(dst)
+		return &arrow.Float64Type{}, nil
+	case schemapb.DataType_VarChar, schemapb.DataType_String:
+		return &arrow.StringType{}, nil
 	case schemapb.DataType_JSON:
-		return typeutil.IsJSONType(dst)
+		return &arrow.StringType{}, nil
+	case schemapb.DataType_Array:
+		elemType, err := convertToArrowDataType(field, true)
+		if err != nil {
+			return nil, err
+		}
+		return arrow.ListOfField(arrow.Field{
+			Name:     "item",
+			Type:     elemType,
+			Nullable: true,
+			Metadata: arrow.Metadata{},
+		}), nil
 	case schemapb.DataType_BinaryVector:
-		return dst == schemapb.DataType_BinaryVector
-	case schemapb.DataType_Float16Vector:
-		return dst == schemapb.DataType_Float16Vector
+		return arrow.ListOfField(arrow.Field{
+			Name:     "item",
+			Type:     &arrow.Uint8Type{},
+			Nullable: true,
+			Metadata: arrow.Metadata{},
+		}), nil
+	case schemapb.DataType_FloatVector:
+		return arrow.ListOfField(arrow.Field{
+			Name:     "item",
+			Type:     &arrow.Float32Type{},
+			Nullable: true,
+			Metadata: arrow.Metadata{},
+		}), nil
+	case schemapb.DataType_Float16Vector, schemapb.DataType_BFloat16Vector:
+		return arrow.ListOfField(arrow.Field{
+			Name:     "item",
+			Type:     &arrow.Uint8Type{},
+			Nullable: true,
+			Metadata: arrow.Metadata{},
+		}), nil
 	default:
-		return false
+		return nil, merr.WrapErrParameterInvalidMsg("unsupported data type %v", dataType.String())
+	}
+}
+
+func ConvertToArrowSchema(schema *schemapb.CollectionSchema) (*arrow.Schema, error) {
+	arrFields := make([]arrow.Field, 0)
+	for _, field := range schema.GetFields() {
+		if typeutil.IsAutoPKField(field) {
+			continue
+		}
+		arrDataType, err := convertToArrowDataType(field, false)
+		if err != nil {
+			return nil, err
+		}
+		arrFields = append(arrFields, arrow.Field{
+			Name:     field.GetName(),
+			Type:     arrDataType,
+			Nullable: true,
+			Metadata: arrow.Metadata{},
+		})
+	}
+	return arrow.NewSchema(arrFields, nil), nil
+}
+
+func isSchemaEqual(schema *schemapb.CollectionSchema, arrSchema *arrow.Schema) error {
+	i := 0
+	for _, field := range schema.GetFields() {
+		if typeutil.IsAutoPKField(field) {
+			continue
+		}
+		arrField := arrSchema.Field(i)
+		if arrField.Name != field.Name {
+			return merr.WrapErrImportFailed(
+				fmt.Sprintf("field name '%s' mis-match with arrow field name '%s", field.Name, arrField.Name))
+		}
+		toArrDataType, err := convertToArrowDataType(field, false)
```

Comment:
We can implement a function named `CanConverted(src *schempb.DataType, dst arrow.DataType) bool` ?

Reply:
for (schemapb.Binary_vector, arrow.Binary), (schemapb.Binary_vector, arrow.Uin8 list), both return true.

Reply:
it's difficult to implement CanConverted(src *schempb.DataType, dst arrow.DataType) bool

Reply:
> for (schemapb.Binary_vector, arrow.Binary), (schemapb.Binary_vector, arrow.Uin8 list), both return true.

what's the expected behavior ?